File: LeetCode/LeetCode Practice Problems/219- Contains Duplicate II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& a, int k) {
        int n = a.size();
        map<int , int> m;
        
        for(int i=0;i<n;i++)
        {
            if(m[a[i]] != 0)
            {
                int diff = (i+1) - m[a[i]];
                if(diff <= k)
                    return true;
            }
            m[a[i]] = i + 1;
        }
        return false;
    }
};

File: LeetCode/LeetCode Practice Problems/219-Contains Duplicate II.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer , Integer> map = new HashMap();
        for(int i=0;i<nums.length;i++) {
            int val = nums[i];
            if(map.containsKey(val)) {
                int prevIdx = map.get(val);
                int diff = i - prevIdx;
                if(diff <= k) {
                    return true;
                }
            }
            map.put(val , i);
        }
        return false;
    }
}
